### ISSUE TITLE
git is needed to run the server

### ISSUE THREAD

**Issue by saludes** [Ptival/PeaCoq#22]
Not really an issue but you might want to mention that `git` is required to **run the server**. I did:

```
peacoq -p 8888
```

and got:

```
peacoq: git: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
```

(Why I didn't have `git` in the first place? Because I built and ran _PeaCoq_ in a docker container, but cloned it previously in a host volume)

**Comment by ztatlock** [comment]
Thanks for the heads up!  I didn't realize git was needed for starting the server.  I added a note in the README.

I've just started looking into packaging PeaCoq for docker.  Are you going to post a container?  Did you come across any other issues?

Thanks for letting us know about the git requirement!

**Comment by saludes** [comment]
> I've just started looking into packaging PeaCoq for docker. Are you going to post a container?

Didn't think of it, but I can pass a  `Dockerfile` if interested.

> Did you come across any other issues?

Yes. I'm using a `haskell` image that has `ghc` version  7.10.1, not 7.10.2. So I needed to patch `HUnit-1.3.0.0` (deleting the `HAS_SOURCE_LOCATIONS` conditionals) to install it.

**Comment by ztatlock** [comment]
> > I've just started looking into packaging PeaCoq for docker. Are you going to post a container?
> 
> Didn't think of it, but I can pass a  `Dockerfile` if interested.

That would extremely helpful!  Thanks :)

> > Did you come across any other issues?
> 
> Yes. I'm using a `haskell` image that has `ghc` version  7.10.1, not 7.10.2. So I needed to patch `HUnit-1.3.0.0` (deleting the `HAS_SOURCE_LOCATIONS` conditionals) to install it.

Is this patch necessary for your docker setup?  If so, we'd love to have it too.

Thanks so much for checking our PeaCoq!  If you don't mind us asking, what are you using it for?

**Comment by saludes** [comment]
> > > I've just started looking into packaging PeaCoq for docker. Are you going to post a container?
> > 
> > Didn't think of it, but I can pass a Dockerfile if interested.
> 
> That would extremely helpful! Thanks :)

Here you are:

```
FROM haskell:latest
RUN cabal update
RUN apt-get update
RUN apt-get install -y git-core wget unzip libpcre3-dev
RUN apt-get install -y coq
RUN git clone https://github.com/Ptival/PeaCoq.git
WORKDIR PeaCoq
RUN ./setup.sh
RUN cabal install
EXPOSE 8888
CMD peacoq -p 8888 
```

> Is this patch necessary for your docker setup? If so, we'd love to have it too.

I realized that now the `haskell:latest` image actually runs `ghc-7.10.2`, so no need for the patch.

> Thanks so much for checking our PeaCoq! If you don't mind us asking, what are you using it for?

My frustrated attempts to run `emacs` on a container and having the windows (via  Xserver) in the host made me to consider other possibilities.

I love the graphical show of possible tactics at a given point. (it needs polishing though).

**Comment by Ptival** [comment]
Should be fixed by 1bc71af8d6f8bc90680cae777a2ac7c537e2b4be.

Thanks for the docker script, I'm adding such things to the repository!
